docker/Dockerfile

```diff
@@ -0,0 +1,76 @@
+FROM golang:1.22.6-alpine3.20 AS build-stage
+
+RUN apk update
+RUN apk add --no-cache bash git
+
+WORKDIR /autograder-server
+
+# Extracts version information from the Git repository and saves it to VERSION.json
+COPY . ./full-copy
+RUN cd full-copy && go run ./cmd/version/main.go --out=VERSION.json && mv VERSION.json ..
+RUN rm -rf ./full-copy
+
+# Copies over only whats nessesary
+COPY ./cmd/ cmd/
+COPY ./internal/ internal/
+COPY ./scripts/ scripts/
+COPY ./testdata/ testdata/
+COPY ./go.* ./
+COPY ./LICENSE ./
+
+COPY ./docker/entrypoint.sh ./
+RUN chmod 0775 entrypoint.sh
+
+# Builds and tests to verify the binaries will always run
+RUN ./scripts/build.sh
+RUN AUTOGRADER__DOCKER__DISABLE='true' ./scripts/run_tests.sh
+RUN rm -rf ./bin
+
+# Clean up unnecessary files and directories to reduce the image size.
+# - Remove 'testdata' directories from Go packages.
+# - Delete test files (*_test.go) from Go packages.
+# - Remove cached ZIP files from Go module downloads.
+# - Clear the Go build cache to further minimize image size.
```

Comment:
The comment has already established we are minimizing size, no need to say "to further minimize image size".